azure/services/scalesets/scalesets.go

```diff
@@ -281,7 +281,7 @@ func (s *Service) patchVMSSIfNeeded(ctx context.Context, infraVMSS *azure.VMSS)
 	// If the VMSS is managed by an external autoscaler, we should patch the VMSS if customData has changed.
 	shouldPatchCustomData := false
 	if s.Scope.HasReplicasExternallyManaged(ctx) {
-		shouldPatchCustomData, err := s.Scope.HasBootstrapDataChanges(ctx)
+		shouldPatchCustomData, err = s.Scope.HasBootstrapDataChanges(ctx)
```

Comment:
Great find @manishsat ! 
Please add some more context to the PR description, as in where & how you found this bug, etc. 
Otherwise looks good to me!
_Holding `lgtm` for more context to be added to the PR description_